src/mkdocs_gallery/scrapers.py

```diff
@@ -192,32 +192,32 @@ def matplotlib_scraper(block, script: GalleryScript, **kwargs):
 
         # save the figures, and populate the srcsetpaths
         try:
-            fig.savefig(image_path, **these_kwargs)
+            fig.savefig(str(image_path), **these_kwargs)
             dpi0 = matplotlib.rcParams["savefig.dpi"]
             if dpi0 == "figure":
                 dpi0 = fig.dpi
             dpi0 = these_kwargs.get("dpi", dpi0)
-            srcsetpaths = {0: image_path}
+            srcsetpaths = {0.0: image_path}
 
             # save other srcset paths, keyed by multiplication factor:
             for mult in srcset_mult_facs:
-                if not (mult == 1):
+                if mult != 1:
```

Comment:
Same here - this is more readable of course, but maybe they had reasons to write the code that way in sphinx-gallery ?
Even if the original code is very strangely constructed IMHO, the less we modify it the more easily we can apply the same fixes than them.

Plus, since we did not port most of their tests yet in our code base (see #12), any modification is a risk of regression without CI safety harness.